crates/topos/src/main.rs

```diff
@@ -63,19 +65,16 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         },
     );
 
-    #[cfg(feature = "tce")]
-    if let options::ToposCommand::Tce(TceCommand {
-        subcommands: Some(TceCommands::Run(ref cmd)),
-        ..
-    }) = args.commands
-    {
+    if let (Some(jaeger_agent), Some(jaeger_service_name)) = (jaeger_agent, jaeger_service_name) {
```

Comment:
Here you removed the filter on the command and put the OTLP argument globally, but I don't want telemetry to be setup for every `topos` commands. Just the one needed, which are sequencer and TCE.
If we stay like this, every command with the environment variables will push traces/logs to our collector.

Reply:
No, if you do not specify it by default otlp is not initialized.

Reply:
Yes but if I have the environment variables define it will enable it on every command. What I want to say is that it is better to activate the telemetry only for some commands and not on the presence of a variable.

Reply:
We have cli commands for sequencer and tce. How many commands will actually we have? 
I think better solution is to error on otlp set for some unwanted command (exceptional case) then to duplicate all the code for telemetry setup.

Reply:
@hadjiszs What do you think?

Reply:
For now, we have those commands that do not need `OTLP`:

- `Tce::status`
- `Tce::push_peer_list`
- `Tce::keys`

In the next PR we will have verifier and a lot more coming in the next iterations.
In fact, the commands that need `OTLP` are `Tce::Run` and `Sequencer::Run` I guess.

To deliver a good user experience with the `topos` cli we should handle everything for the user. If the user is executing a command with some variables defined, we should not fail the command and ask him to do something if they are completely unneeded.

Having the `OTLP` on the root options means that they display on every command too, misleading the user that they can define them when they are not used.

Reply:
Ok, fair enough. I will set them up per sequencer and tce command.